UnitTests/AppAuthTV/OIDTVTokenRequestTests.m

```diff
@@ -0,0 +1,196 @@
+    @"urn:ietf:params:oauth:grant-type:device_code";
+
+@implementation OIDTVTokenRequestTests
+
+- (NSDictionary<NSString *, NSString *> *)bodyParametersFromURLRequest:
+    (NSURLRequest *)URLRequest {
+  NSString *bodyString = [[NSString alloc] initWithData:URLRequest.HTTPBody
+                                               encoding:NSUTF8StringEncoding];
+  NSArray<NSString *> *bodyParameterStrings =
+      [bodyString componentsSeparatedByString:@"&"];
+
+  NSMutableDictionary<NSString *, NSString *> *bodyParameters =
+      [[NSMutableDictionary alloc] init];
+
+  for (NSString *paramString in bodyParameterStrings) {
+    NSArray<NSString *> *components =
+        [paramString componentsSeparatedByString:@"="];
+
+    if (components.count == 2) {
+      bodyParameters[components[0]] = components[1];
+    }
+  }
+
+  return bodyParameters;
+}
+
+- (OIDTVServiceConfiguration *)testServiceConfiguration {
+  NSURL *tokenEndpoint = [NSURL URLWithString:kTestTokenEndpoint];
+  NSURL *TVAuthorizationEndpoint = [NSURL URLWithString:kTestTVAuthorizationEndpoint];
+
+  // Pass in an empty authorizationEndpoint since only the
+  // TVAuthorizationEndpoint and tokenEndpoint are used for the TV
+  // authentication flow.
+  OIDTVServiceConfiguration *configuration = [[OIDTVServiceConfiguration alloc]
+      initWithAuthorizationEndpoint:[[NSURL alloc] initWithString:@""]
+            TVAuthorizationEndpoint:TVAuthorizationEndpoint
+                      tokenEndpoint:tokenEndpoint];
+  return configuration;
+}
+
+- (OIDTVTokenRequest *)testTokenRequest {
+  OIDTVServiceConfiguration *service = [self testServiceConfiguration];
+  return [[OIDTVTokenRequest alloc]
+      initWithConfiguration:service
+                 deviceCode:kDeviceCodeValue
+                   clientID:kTestClientID
+               clientSecret:kTestClientSecret
+       additionalParameters:@{kTestAdditionalParameterKey : kTestAdditionalParameterValue}];
+}
+
+/*! @brief Tests the initializer
+*/
+- (void)testInitializer {
+  OIDTVTokenRequest *request = [self testTokenRequest];
+  NSURL *requestTVAuthorizationEndpoint =
+  ((OIDTVServiceConfiguration *)request.configuration).TVAuthorizationEndpoint;
+  
+  XCTAssertEqualObjects(requestTVAuthorizationEndpoint,
+                        [self testServiceConfiguration].TVAuthorizationEndpoint);
+  XCTAssertEqualObjects(request.deviceCode, kDeviceCodeValue);
+  XCTAssertEqualObjects(request.grantType, kOIDTVDeviceTokenGrantType);
+  XCTAssertEqualObjects(request.clientID, kTestClientID);
+  XCTAssertEqualObjects(request.clientSecret, kTestClientSecret);
+  XCTAssertEqualObjects(request.additionalParameters,
+                        @{kTestAdditionalParameterKey:kTestAdditionalParameterValue});
+}
+
+/*! @brief Tests the @c NSCopying implementation by round-tripping an instance through the copying
+* process and checking to make sure the source and destination both contain the @c deviceCode.
```

Comment:
the second line should be indented 4 characters past the @ of the previous line. here and elsewhere